test/request.test.js

```diff
@@ -183,28 +189,57 @@ describe("request", () => {
     responseStub.headers = {
       "content-encoding": "gzip"
     };
-    httpsStub.request.firstCall.args[1](responseStub);
+    requestStub.emit("response", responseStub);
     return promise.then(fail, error => {
       assert.equal(error.message, "incorrect header check");
     });
   });
 
-  it("should reject the promise on socket timeout", () => {
-    requestStub.abort = sinon.stub();
-    const promise = request();
-    requestStub.emit("timeout");
-    return promise.then(fail, error => {
-      assert.equal(error.message, "socket timeout");
-      assert(requestStub.abort.calledOnce);
-    });
+  it("should reject the promise on connection timeout", done => {
+    const timeout = 100;
+    const host = "example.org";
+
+    request({ timeout, host })
+      .catch(error => {
```

Comment:
Can you try breaking the code, aka not having a timeout and seeing if the test passes?
Or does mocha enforces some assertion being run?

Reply:
I've added a check that would break the test and thanks to reviewing myself discovered an issue in a different test and implementation